src/main/kotlin/com/exactpro/th2/codec/configuration/Configuration.kt

```diff
@@ -13,81 +13,49 @@
 
-enum class ProcessorType { CUMULATIVE, SEQUENTIAL }
-
-class Configuration() {
-    var codecClassName: String? = null
-
-    var codecParameters: Map<String, String>? = null
-
-    var decoderInputAttribute: String = "decoder_in"
-    var decoderOutputAttribute: String = "decoder_out"
-
-    var encoderInputAttribute: String = "encoder_in"
-    var encoderOutputAttribute: String = "encoder_out"
-
-    var generalDecoderInputAttribute: String = "general_decoder_in"
-    var generalDecoderOutputAttribute: String = "general_decoder_out"
-
-    var generalEncoderInputAttribute: String = "general_encoder_in"
-    var generalEncoderOutputAttribute: String = "general_encoder_out"
-
-    var decodeProcessorType: ProcessorType = ProcessorType.CUMULATIVE
+class Configuration {
+    var codecSettings: IPipelineCodecSettings? = null
 
     companion object {
-
-        fun create(commonFactory : CommonFactory, sailfishCodecParamsPath: String?) : Configuration{
-
-            val configuration = commonFactory.getCustomConfiguration(Configuration::class.java)
-            configuration.codecParameters = readSailfishParameters(sailfishCodecParamsPath)
-            return configuration
-        }
-
-
-        private fun readSailfishParameters(sailfishCodecParamsPath: String?): Map<String, String> {
-            if (StringUtils.isBlank(sailfishCodecParamsPath)) {
-                return mapOf()
-            }
-            val codecParameterFile = Paths.get(sailfishCodecParamsPath)
-            if (!Files.exists(codecParameterFile)) {
-                return mapOf()
-            }
-            try {
-                return OBJECT_MAPPER.readValue(
-                        newInputStream(codecParameterFile),
-                        object : TypeReference<LinkedHashMap<String, String>>() {}
-                )
-            } catch (exception: Exception) {
-                when (exception) {
-                    is IOException,
-                    is JsonParseException,
-                    is JsonMappingException -> {
-                        throw ConfigurationException("could not parse '$sailfishCodecParamsPath' file", exception)
+        const val DECODER_INPUT_ATTRIBUTE: String = "decoder_in"
+        const val DECODER_OUTPUT_ATTRIBUTE: String = "decoder_out"
+        const val ENCODER_INPUT_ATTRIBUTE: String = "encoder_in"
+        const val ENCODER_OUTPUT_ATTRIBUTE: String = "encoder_out"
+        const val GENERAL_DECODER_INPUT_ATTRIBUTE: String = "general_decoder_in"
+        const val GENERAL_DECODER_OUTPUT_ATTRIBUTE: String = "general_decoder_out"
+        const val GENERAL_ENCODER_INPUT_ATTRIBUTE: String = "general_encoder_in"
+        const val GENERAL_ENCODER_OUTPUT_ATTRIBUTE: String = "general_encoder_out"
+
+        fun create(commonFactory: CommonFactory, settingsPath: String?): Configuration {
+            return commonFactory.getCustomConfiguration(Configuration::class.java).apply {
+                codecSettings = settingsPath?.run {
```

Comment:
It is the main point. The other ones can wait

Reply:
Okay, but it's gonna require some fiddling with reflection, because settings are arbitrary beans

Reply:
I believe there is another way. Let me check something, please

Reply:
I thought we can use `ObjectMapper.readerForUpdating` but unfortunately it behaves differently than I expect...
Maybe instead of using reflection, we will just change the priority of parameters? I mean if parameters are set in the custom config we will use them instead of loading parameters from the file?